toolbox/relnotes/main.go

```diff
@@ -644,15 +644,28 @@ func patchRelease(f *os.File, info *ReleaseInfo) {
 }
 
 // extractReleaseNoteFromPR tries to fetch release note from PR body, otherwise uses PR title.
-func extractReleaseNoteFromPR(pr *github.Issue) string {
+func extractReleaseNoteFromPR(pr *github.Issue, actionRequired bool) string {
+	var releaseNote string
 	// Regexp Example:
 	// This regexp matches the release note section in Kubernetes pull request template:
 	// https://github.com/kubernetes/kubernetes/blob/master/.github/PULL_REQUEST_TEMPLATE.md
-	re, _ := regexp.Compile("```release-note\r\n(.+)\r\n```")
+	re := regexp.MustCompile("(?s)```release-note\r\n(.+)\r\n```")
 	if note := re.FindStringSubmatch(*pr.Body); note != nil {
-		return note[1]
+		releaseNote = strings.TrimSpace(note[1])
+	} else {
+		// Use PR title if unable to find release note
+		releaseNote = strings.TrimSpace(*pr.Title)
+	}
+	// We do this for action required PRs only, to avoid removing "action required" message from
+	// some PR that accidentally gets marked as non-action-required PR.
+	if actionRequired {
+		re = regexp.MustCompile("(?is)^\\[*action required\\]*:* *(.+)$")
+		if note := re.FindStringSubmatch(releaseNote); note != nil {
+			releaseNote = strings.TrimSpace(note[1])
```

Comment:
Just a question: I'm not very clear about the above regexp, will this code block make the `action required` words be removed from the release note finally?

Reply:
Yes, `note` returned from `FindStringSubmatch` is a slice of match and submatches. `note[0]` would be the whole release note; and `note[1]` would be the `(.+)` part in the regexp, which doesn't include the "action required" text.

Reply:
Thanks!